pkgs/development/python-modules/spherogram/default.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  cython,
+  networkx,
+  decorator,
+  knot-floer-homology,
+  snappy-manifolds,
+  snappy-15-knots,
+  include15CrossingKnots ? true,
+}:
+
+buildPythonPackage rec {
+  pname = "spherogram";
+  version = "2.3";
+
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "3-manifolds";
+    repo = "spherogram";
+    tag = "${version}_as_released";
+    hash = "sha256-uqc+3xS4xulXR0tZlNuyC5Zz5OztR6c4PZWpsvU+4Pw=";
+  };
+
+  build-system = [
+    setuptools
+    cython
+  ];
+
+  dependencies = [
+    networkx
+    decorator
+    snappy-manifolds
+    knot-floer-homology
+  ] ++ lib.optionals include15CrossingKnots [ snappy-15-knots ];
```

Comment:
Can you ask upstream to declare this as on optional dependency and use `optional-dependencies`?

Reply:
No need to patch it in Nixpkgs. But upstream uses setup.py, where it's called `extra_requires`.

Reply:
Just add `optional-dependencies` already.